### PR TITLE
checkSignatures path 3 - signatures length 1

### PR DESCRIPTION
**Pull request by denis-bogdanas**
1. Gnosis: checkSignatures branch 3(v>26): specs for this case, only for one signature.

2. Added lemmas requires for these specs. Function byteStack2IntList() moved to evm-semantics/edsl.md
3. Changed gnosis bytecode to a version with some methods public. No other changes.
5. Minor fixes in kprove tutorial.

### CODE REVIEW

**Comment by denis-bogdanas** [comment]
jenkins: test this please

**Comment by daejunpark** [comment]
do you still want to merge, or just keep it as a branch, in case we switch to the new memory model?

**Comment by denis-bogdanas** [comment]
@daejunpark Please merge. Branch contains too many changes to keep it desynchronzied from master. I added some general expression simplification lemmas that required small changes to bihu for exmple. There are also updates to kprove tutorial.

File: resources/lemmas.md [hunk 1]
```diff
@@ -176,13 +176,6 @@ Code sugar to represent byte arrays with concrete size but symbolic data.
     rule keccak(WS) => keccakIntList(byteStack2IntList(WS))
       requires ( notBool #isConcrete(WS) )
        andBool ( #sizeWordStack(WS) ==Int 32 orBool #sizeWordStack(WS) ==Int 64 )
-
-    // inverse of intList2ByteStack of edsl.md
-    syntax IntList ::= byteStack2IntList ( WordStack )       [function]
-                     | byteStack2IntList ( WordStack , Int ) [function]
-    rule byteStack2IntList ( WS ) => byteStack2IntList ( WS , #sizeWordStack(WS) /Int 32 ) requires #sizeWordStack(WS) %Int 32 ==Int 0
-    rule byteStack2IntList ( WS , N ) => #asWord ( WS [ 0 .. 32 ] ) byteStack2IntList ( #drop(32, WS) , N -Int 1 ) requires N >Int 1
-    rule byteStack2IntList ( WS , 1 ) => #asWord ( WS [ 0 .. 32 ] ) .IntList
```
Comment (daejunpark) [inline resources/lemmas.md]: why did you drop this?

Comment (daejunpark) [reply resources/lemmas.md]: ah nvm. you moved to another place.

**Comment by daejunpark** [comment]
this is not used yet here, right?

https://github.com/kframework/evm-semantics/commit/5d3bbb6d9dd3a2913201071937cb3f00d66832b0

**Comment by denis-bogdanas** [comment]
Yes split is not used. I use it in the following PR.

**Pull request merged**